plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java

```diff
@@ -533,6 +536,25 @@ public boolean isForceTransform(DBCSession session, SQLQuery sqlQuery) {
         return false;
     }
 
+    public boolean isLimitApplicableTo(SQLQuery query) {
+        boolean hasNextValExpr = false;
+        try {
+            Statement statement = SQLSemanticProcessor.parseQuery(this.sqlDialect, query.getText());
+            if (statement instanceof Select) {
+                SelectBody selectBody = ((Select) statement).getSelectBody();
+                if (selectBody instanceof PlainSelect) {
+                    hasNextValExpr = ((PlainSelect) selectBody).getSelectItems().stream().anyMatch(
+                        item -> (item instanceof SelectExpressionItem) 
+                            && (((SelectExpressionItem) item).getExpression() instanceof NextValExpression)
+                    );
```

Comment:
Does ANY nextval expression in SELECT ... restricts setLimit?
Is it possible to run 
`select xxx.nextval, t.col1 FROM table t`

I think query must not operate with tables.